### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by corbin-coleman**
Add the CODEOWNERS file to auto assign reviewers for the CHANGLOG and components/cli.

Signed-off-by: Corbin <corbin.coleman@docker.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# GitHub code owners
+# See https://help.github.com/articles/about-codeowners/
+#
+# KEEP THIS FILE SORTED. Order is important. Last match takes precedence
+
+CHANGELOG-ee.md @mstanleyjones @joaofnfernandes
```
Comment (thaJeztah) [inline .github/CODEOWNERS]: this is the `ce` repo, so don't think this file will be there?

Comment (corbin-coleman) [reply .github/CODEOWNERS]: Don't know how I missed that. Got it fixed. Thanks!

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# GitHub code owners
+# See https://help.github.com/articles/about-codeowners/
+#
+# KEEP THIS FILE SORTED. Order is important. Last match takes precedence
+
+CHANGELOG.md @mstanleyjones @joaofnfernandes
```
Comment (andrewhsu) [inline .github/CODEOWNERS]: replace joao with @gbarr01

**Comment by seemethere** [comment]
Do we still need this @docker/release-engineering?

**Comment by jose-bigio** [comment]
Following up with what @seemethere  wrote do you want to close this due to inactivity or because there is no longer a need?

**Comment by thaJeztah** [comment]
looks outdated by now, so I suggest closing (at least for now)

**Comment by jose-bigio** [comment]
Closing for now

**Pull request closed without merging**